mirrord/cli/src/operator/session.rs

```diff
@@ -0,0 +1,102 @@
+use kube::{api::DeleteParams, core::Status, Api};
+use mirrord_operator::client::{session_api, OperatorApiError, OperatorOperation};
+use mirrord_progress::{Progress, ProgressTracker};
+use tracing::error;
+
+use crate::{error::CliError, Result};
+
+/// Prepares progress and kube api for use in the operator session commands.
+#[tracing::instrument(level = "trace", ret)]
+async fn operator_session_prepare() -> Result<(
+    ProgressTracker,
+    Api<mirrord_operator::crd::SessionCrd>,
+    ProgressTracker,
+)> {
+    let progress = ProgressTracker::from_env("Operator session action");
+
+    let session_api = session_api(None)
+        .await
+        .inspect_err(|fail| error!("Failed to even get sessio_api {fail:?}!"))?;
+
+    let sub_progress = progress.subtask("preparing to execute session operation...");
+
+    Ok((progress, session_api, sub_progress))
+}
+
+/// Handles the cleanup part of progress after an operator session command.
+#[tracing::instrument(level = "trace", ret)]
+fn operator_session_finished(
+    result: Option<Status>,
+    mut sub_progress: ProgressTracker,
+    mut progress: ProgressTracker,
+) -> Result<()> {
+    if let Some(status) = result {
+        sub_progress.success(Some(&format!(
+            "session operation completed with {status:?}"
```

Comment:
Don't show debug impl, maybe just `... completed successfuly (status code ...)` or `... failed (status code ...)`

Reply:
Changed to give out more messages based on status.